src/backend/distributed/operations/shard_split.c

```diff
@@ -0,0 +1,843 @@
+		{
+			char *sourceRelationName = get_rel_name(relationId);
+			char *colocatedRelationName = get_rel_name(colocatedTableId);
+
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot %s of '%s', because it "
+								   "is a partitioned table",
+								   SplitOperationName[splitOperation],
+								   colocatedRelationName),
+							errdetail("In colocation group of '%s', a partitioned "
+									  "relation exists: '%s'. Citus does not support "
+									  "%s of partitioned tables.",
+									  sourceRelationName,
+									  colocatedRelationName,
+									  SplitOperationName[splitOperation])));
+		}
+	}
+
+	/* check shards with inactive placements */
+	List *colocatedShardList = ColocatedShardIntervalList(sourceShard);
+	foreach(colocatedShardCell, colocatedShardList)
+	{
+		ShardInterval *shardInterval = (ShardInterval *) lfirst(colocatedShardCell);
+		uint64 shardId = shardInterval->shardId;
+		ListCell *shardPlacementCell = NULL;
+
+		List *shardPlacementList = ShardPlacementListWithoutOrphanedPlacements(shardId);
+		foreach(shardPlacementCell, shardPlacementList)
+		{
+			ShardPlacement *placement = (ShardPlacement *) lfirst(shardPlacementCell);
+			if (placement->shardState != SHARD_STATE_ACTIVE)
+			{
+				char *relationName = get_rel_name(shardInterval->relationId);
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("cannot %s %s because relation "
+									   "\"%s\" has an inactive shard placement "
+									   "for the shard %lu",
+									   SplitOperationName[splitOperation],
+									   SplitTargetName[splitOperation],
+									   relationName, shardId),
+								errhint("Use master_copy_shard_placement UDF to "
+										"repair the inactive shard placement.")));
+			}
+		}
+	}
+}
+
+
+/*
+ * Exteded checks before we decide to split the shard.
+ * When all consumers (Example : ISOLATE_TENANT_TO_NEW_SHARD) directly call 'SplitShard' API,
+ * this method will be merged with 'ErrorIfCannotSplitShard' above.
+ */
+static void
+ErrorIfCannotSplitShardExtended(SplitOperation splitOperation,
+								ShardInterval *shardIntervalToSplit,
+								List *shardSplitPointsList,
+								List *nodeIdsForPlacementList)
+{
+	CitusTableCacheEntry *cachedTableEntry = GetCitusTableCacheEntry(
+		shardIntervalToSplit->relationId);
+
+	/* Perform checks common to both blocking and non-blocking Split API here. */
+	if (!IsCitusTableTypeCacheEntry(cachedTableEntry, HASH_DISTRIBUTED))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("Cannot %s %s as operation "
+							   "is only supported for hash distributed tables.",
+							   SplitOperationName[splitOperation],
+							   SplitTargetName[splitOperation])));
+	}
+
+	if (extern_IsColumnarTableAmTable(shardIntervalToSplit->relationId))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("Cannot %s %s as operation "
+							   "is not supported for Columnar tables.",
```

Comment:
We should support columnar. It's very common to use distributed tables in combination with columnar.

Reply:
I am assuming you are referring specifically to supporting columnar only for blocking split right?

Reply:
We should support columnar in both blocking and non-blocking cases as we do for shard moves. While logical replication does not really do anything for columnar after the initial data copy, setting it a subscription actually works. All writes fail while the columnar table is part of a publication.

Reply:
@marcocitus 
"While logical replication does not really do anything for columnar, setting it a subscription actually works."
So you mean data will replicate but writes will be blocked ? The customer specifically asked for non-blocking and is not expecting downtime (maybe they would not split had they known we will block) but we let them proceed with downtime? Isn't that bad?

CC: @saawasek

Reply:
We throw an error unless they explicitly specify `shard_transfer_mode := 'force_logical'` , because columnar tables also do not have replica identities.

Reply:
I assume you'll fix this along with partitioning in the next PR.

Reply:
@marcocitus Yes, Next PR will fix columnar and partitioning support.